std/typecons.d

```diff
@@ -6896,3 +6896,418 @@ unittest
         ubyte, ubyte, T3, T3,
     );
 }
+
+private enum isNullable(T) = is(typeof(T.init is null));
```

Comment:
Yes, but it's outside the scope of this PR. They can be moved later. (I think std.allocator also has its own `isCopyable` template)